x/auth/keeper/keeper.go

```diff
@@ -182,7 +184,7 @@ func (ak AccountKeeper) GetSequence(ctx context.Context, addr sdk.AccAddress) (u
 // NextAccountNumber returns and increments the global account number counter.
 // If the global account number is not set, it initializes it with value 0.
 func (ak AccountKeeper) NextAccountNumber(ctx context.Context) uint64 {
-	n, err := ak.AccountNumber.Next(ctx)
+	n, err := ak.AccountsModKeeper.NextAccountNumber(ctx)
```

Comment:
personal preference: It would be good to deprecate this method so that there is only one way to get new account numbers in the future